app/controllers/api/flavors_controller.rb

```diff
@@ -1,4 +1,30 @@
 module Api
   class FlavorsController < BaseController
+    def create_resource(_type, _id, data)
+      attrs = validate_flavor_attrs(data)
+      task_id = Flavor.create_flavor_queue(User.current_user.id, EmsCloud.find_by_id(attrs['ems']['id']), attrs.deep_symbolize_keys)
+      action_result(true, 'Creating Flavor', :task_id => task_id)
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    def delete_resource(type, id, _data = {})
+      flavor = resource_search(id, type, collection_class(:flavors))
+      raise "Delete not supported for #{flavor_ident(flavor)}" unless flavor.respond_to?(:delete_flavor_queue)
```

Comment:
is this supposed to be `delete_in_provider_queue`? it's checking a different method than is used below

Reply:
Why don't all flavors respond to this? How does the user know which ones can be deleted? Is it possible that flavors could utilize the supports_feature? feature?

Reply:
All flavors support delete. I should delete this line of code.